src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,364 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Shared.Extensions;
+using Microsoft.CodeAnalysis.CSharp.UseParameterNullChecking;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.UseParameterNullChecking
+{
+    public partial class UseParameterNullCheckingTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
```

Comment:
This should use the new test framework.

Reply:
Could you please provide an example of usage of the new framework? I actually just copied all the corresponding files from `UseIsNullCheckForCastAndEqualityOperator` as a basis for this.

Reply:
@RikkiGibson https://github.com/dotnet/roslyn/blob/main/src/Analyzers/CSharp/Tests/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs